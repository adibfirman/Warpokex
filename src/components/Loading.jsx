import React, {useState, useEffect} from 'react' 
import {bool, number} from 'prop-types'

import { Div } from "atomize";

export default function Loading({loading, duration}) {
    const [animate, setAnimate] = useState(loading) 

    useEffect(() => {
        setInterval(setAnimate(!animate), duration) 
    }, [animate, duration])

    return(
        <Div
          h="3rem"
          w="3rem"
          m={{ b: "4rem" }}
          rounded="lg"
          bg="brand200"
          style={{
            transform: animate
              ? "translateX(20rem)"
              : "translateX(0)"
          }}
          transition
        />
    )
}

Loading.propTypes = {
    loading: bool,
    duration: number,
}

Loading.defaultProps = {
    loading: false,
    duration: 1000,
}
