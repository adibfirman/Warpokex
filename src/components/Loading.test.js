import React from 'react'
import Loading from 'components/Loading'

test('should render loading component correctly', async () => {
  const { container } = render(<Loading loading={true} duration={3000} />)
  expect(container).toMatchSnapshot()
})
