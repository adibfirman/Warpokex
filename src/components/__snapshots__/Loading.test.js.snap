// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render loading component correctly 1`] = `
<div>
  <div
    class="sc-bdVaJa HAhlL"
    style="position: absolute; top: 50vh; bottom: 50vh; transform: translateX(0);"
  />
</div>
`;
